lib/common/tests/xml/pcmk__xml_is_name_start_char_test.c

```diff
@@ -0,0 +1,276 @@
+        assert_name_start_char(c, true);
+    }
+}
+
+static void
+unicode_0x300_to_0x36F(void **state)
+{
+    for (int c = 0x300; c <= 0x36F; c++) {
+        assert_name_start_char(c, false);
+    }
+}
+
+static void
+unicode_0x370_to_0x37D(void **state)
+{
+    for (int c = 0x370; c <= 0x37D; c++) {
+        assert_name_start_char(c, true);
+    }
+}
+
+static void
+unicode_0x37E(void **state)
+{
+    assert_name_start_char(0x37E, false);
+}
+
+static void
+unicode_0x37F_to_0x1FFF(void **state)
+{
+    for (int c = 0x37F; c <= 0x1FFF; c++) {
+        assert_name_start_char(c, true);
+    }
+}
+
+static void
+unicode_0x2000_to_0x200B(void **state)
+{
+    for (int c = 0x2000; c <= 0x200B; c++) {
+        assert_name_start_char(c, false);
+    }
+}
+
+static void
+unicode_0x200C_to_0x200D(void **state)
+{
+    for (int c = 0x200C; c <= 0x200D; c++) {
+        assert_name_start_char(c, true);
+    }
+}
+
+static void
+unicode_0x200E_to_0x206F(void **state)
+{
+    for (int c = 0x200E; c <= 0x206F; c++) {
+        assert_name_start_char(c, false);
+    }
+}
+
+static void
+unicode_0x2070_to_0x218F(void **state)
+{
+    for (int c = 0x2070; c <= 0x218F; c++) {
+        assert_name_start_char(c, true);
+    }
+}
+
+static void
+unicode_0x2190_to_0x2BFF(void **state)
+{
+    for (int c = 0x2190; c <= 0x2BFF; c++) {
+        assert_name_start_char(c, false);
+    }
+}
+
+static void
+unicode_0x2C00_to_0x2FEF(void **state)
+{
+    for (int c = 0x2C00; c <= 0x2FEF; c++) {
+        assert_name_start_char(c, true);
+    }
+}
+
+static void
+unicode_0x2FF0_to_0x3000(void **state)
+{
+    for (int c = 0x2FF0; c <= 0x3000; c++) {
+        assert_name_start_char(c, false);
+    }
+}
+
+static void
+unicode_0x3001_to_0xD7FF(void **state)
+{
+    for (int c = 0x3001; c <= 0xD7FF; c++) {
+        assert_name_start_char(c, true);
+    }
+}
+
+static void
+unicode_0xD800_to_0xF8FF(void **state)
+{
+    for (int c = 0xD800; c <= 0xF8FF; c++) {
+        assert_name_start_char(c, false);
+    }
+}
+
+static void
+unicode_0xF900_to_0xFDCF(void **state)
+{
+    for (int c = 0xF900; c <= 0xFDCF; c++) {
+        assert_name_start_char(c, true);
+    }
+}
+
+static void
+unicode_0xFDD0_to_0xFDEF(void **state)
+{
+    for (int c = 0xFDD0; c <= 0xFDEF; c++) {
+        assert_name_start_char(c, false);
+    }
+}
+
+static void
+unicode_0xFDF0_to_0xFFFD(void **state)
+{
+    for (int c = 0xFDF0; c <= 0xFFFD; c++) {
+        assert_name_start_char(c, true);
+    }
+}
+
+static void
+unicode_0xFFFE_to_0xFFFF(void **state)
+{
+    for (int c = 0xFFFE; c <= 0xFFFF; c++) {
+        assert_name_start_char(c, false);
+    }
+}
+
+static void
+unicode_0x10000_to_0xEFFFF(void **state)
+{
+    for (int c = 0x10000; c <= 0xEFFFF; c++) {
+        assert_name_start_char(c, true);
+    }
+}
+
+static void
+unicode_0xF0000_to_0x10FFFF(void **state)
+{
+    for (int c = 0xF0000; c <= 0x10FFFF; c++) {
+        assert_name_start_char(c, false);
+    }
+}
+
+PCMK__UNIT_TEST(NULL, NULL,
```

Comment:
Did I HAVE to test every single character? No. But there's no noticeable slowdown at all when unit-testing on my machine. So, might as well for completeness, instead of picking random characters in each range.